packages/expo-dev-launcher/src/DevLauncher.ts

```diff
@@ -0,0 +1,55 @@
+import { ErrorUtils } from 'react-native';
+
+// Similar interface to the one used in expo modules.
+type CodedError = Error & { code?: string };
+
+let errorHandlerWasRegistered = false;
+
+const possibleSolutions = `Some possible solutions:
+- Make sure you're using the newest available version of this development client.
+- Make sure you're running a compatible version of your JavaScript code.
+- If you've installed a new library recently, you may need to make a new development client build.`;
+
+function customizeUnavailableMessage(error: CodedError) {
+  error.message += '\n\n' + possibleSolutions;
+}
+
+function customizeModuleIsMissingMessage(error: Error) {
+  error.message = `Your JavaScript code tried to access a native module that doesn't exist in this development client. 
+
+${possibleSolutions}`;
+}
+
+function customizeError(error: Error | CodedError) {
+  if ('code' in error) {
+    // It's a CodeError from expo modules
+    switch (error.code) {
+      case 'ERR_UNAVAILABLE': {
+        customizeUnavailableMessage(error);
```

Comment:
I'm not sure we want to add the message with the same possible solutions here, because this type of error can also be thrown when the method is simply not supported on the current platform, in which case we will send people down a rabbithole of possible solutions that aren't going to work. Maybe we could just improve the default message that's created by `UnavailabilityError` instead of trying to monkeypatch it here?

Reply:
I'm not a big fan of improving the default message, cause our suggestions are only valid in the dev-client context. I'm guessing that it would be better to add another solution to the list in that case. Something like: `Make sure that the method is available on the current platform`.

Reply:
OK, that seems reasonable to me. Could we make that one the first suggestion in that case?

Reply:
done 😉